compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -396,7 +396,12 @@ class Typer extends Namer
       }
       else if (name.toTermName == nme.ERROR)
         UnspecifiedErrorType
-      else
+      else if (ctx.owner.isConstructor && ctx.mode.is(Mode.InSuperCall) &&
```

Comment:
Doesn't `ctx.mode.is(Mode.InSuperCall)` implies `ctx.owner.isConstructor`?

Reply:
It might be a method super call with `super.xys`.